app/assets/javascripts/esbuild.js

```diff
@@ -0,0 +1,67 @@
+// See: https://esbuild.github.io/plugins/#webassembly-plugin
+
+const esbuild = require("esbuild");
+const path = require("node:path");
+const fs = require("node:fs");
+const { argv } = require("node:process");
+
+let wasmPlugin = {
+  name: "wasm",
+
+  setup(build) {
+    build.onResolve({ filter: /\.wasm$/ }, (args) => {
+      if (args.namespace === "wasm-stub") {
+        return {
+          path: args.path,
+          namespace: "wasm-binary",
+        };
+      }
+
+      if (args.resolveDir === "") {
+        return;
+      }
+
+      return {
+        path: path.isAbsolute(args.path)
+          ? args.path
+          : path.join(args.resolveDir, args.path),
+        namespace: "wasm-stub",
+      };
+    });
+
+    build.onLoad({ filter: /.*/, namespace: "wasm-stub" }, async (args) => {
+      return {
+        contents: `import wasm from ${JSON.stringify(args.path)};
+        export default (imports) => {
+          const wasmModule = new WebAssembly.Module(wasm);
+          const wasmInstance = new WebAssembly.Instance(wasmModule, imports);
+          return wasmInstance.exports;
+        };`,
+      };
+    });
+
+    build.onLoad({ filter: /.*/, namespace: "wasm-binary" }, async (args) => {
+      return {
+        contents: await fs.promises.readFile(args.path),
+        loader: "binary",
+      };
+    });
+  },
+};
+
+esbuild
+  .build({
+    logLevel: "warning",
+    bundle: true,
+    minify: true,
```

Comment:
```suggestion
    minify: false,
```
IMO there's no need to minify code that we're running on the server. Better to keep all the original function/variable names so that error stacktraces are more useful?

Reply:
Minifying reduces the final file size from 11.3MB to 7MB. This may slightly improve memory usage? (and the loading speed by like two nanoseconds 😄)

And since that minified code is run in miniracer env we (currently) don't get stack traces anyway. This the string that gets returned when an error happens:

```
console.error("[THEME 1 'marks'] Compile error: Uncaught ERROR TEST at theme-transpiler.js:199689:4 (discourse/components/my-component.gjs)");
```